### ISSUE TITLE
Use `repo.saltstack.com` where appropriate/possible

### ISSUE THREAD

**Issue by eliasp** [saltstack-formulas/salt-formula#180]
Currently, the following distributions/releases are covered by https://repo.saltstack.com:
- [ ] Debian
  - [ ] Jessie (8)
- [ ] Ubuntu
  - [ ] Precise Pangolin (12.04)
  - [ ] Trusty Tahr (14.04)
- [ ] RHEL/CentOS
  - [ ] RHEL/CentOS 5
  - [ ] RHEL/CentOS 6
  - [ ] RHEL/CentOS 7

`salt.pkgrepo` should make use of `repo.saltstack.com` where appropriate/possible.

I might take care of providing a PR for Ubuntu. Can't promise yet to add support for other distributions as well.

**Comment by TheBigBear** [comment]
+1

**Comment by rhertzog** [comment]
Once  #234 is merged you should be able to cross the Debian and Ubuntu items in the todo list. Note that Debian 7 Wheezy is available on repo.saltstack.com as well.

**Comment by jbouse** [comment]
Currently the salt-formula does not properly work on a Raspberry Pi running Raspbian... 

- First issue is that the salt.pkgrepo state tries to include salt.pkgrepo.raspbian because grains['os'] returns Rasbian but grains['os_family'] returns Debian. This was easily overcome in some crude testing by symlinking salt/pkgrepo/debian as salt/pkgrepo/raspbian but that lead to the second issue.
- Second issue is that the salt/map.jinja hard codes the 'amd64' arch into the pkgrepo setting value rather than using grains['osarch'] which returns the appropriate 'armhr'. 

Adding the symlink along with the following patch appears to work for Debian and Raspbian but would obviously break for Ubuntu systems that are also Debian os_family.

```
diff --git a/salt/map.jinja b/salt/map.jinja
index 2da0b0b..8cdccd0 100644
--- a/salt/map.jinja
+++ b/salt/map.jinja
@@ -12,8 +12,8 @@ that differ from whats in defaults.yaml
 {% set os_family_map = salt['grains.filter_by']({
     'Debian':  {
       'pkgrepo': 'deb http://repo.saltstack.com/apt/' +
-      salt['grains.get']('os')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/amd64/latest ' + salt['grains.get']('oscodename') + ' main',
-      'key_url': 'https://repo.saltstack.com/apt/' + salt['grains.get']('os')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/amd64/latest/SALTSTACK-GPG-KEY.pub',
+      salt['grains.get']('os_family')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/' + salt['grains.get']('osarch') + '/latest ' + salt['grains.get']('oscodename') + ' main',
+      'key_url': 'https://repo.saltstack.com/apt/' + salt['grains.get']('os_family')|lower + '/' + salt['grains.get']('osmajorrelease', osrelease) + '/' + salt['grains.get']('osarch') + '/latest/SALTSTACK-GPG-KEY.pub',
       'libgit2': 'libgit2-22',
       'gitfs': {
         'pygit2': {

```

**Comment by babilen** [comment]
@jbouse thank you. Could you open a PR for the `grains['osarch']` fix? That would be most appreciated!

**Comment by jbouse** [comment]
@babilen Yeah, I'm fine putting a PR in... I want to find a solution that doesn't break Ubuntu while fixing for Rasbian. As soon as I do that I'll get the PR submitted from my fork.

**Comment by rhertzog** [comment]
On Fri, 12 Aug 2016, Jeremy T. Bouse wrote:
> - Second issue is that the salt/map.jinja hard codes the 'amd64' arch
> into the pkgrepo setting value rather than using grains['osarch'] which
> returns the appropriate 'armhr'. 

That's because the ubuntu/debian repositories only have "amd64"
directories and no directories for other architectures AFAIR. That's
not a problem because the packages are "Arch: all" and thus can
be used on any architecture.

Cheers,
-- 
Raphaël Hertzog ◈ Debian Developer

Support Debian LTS: http://www.freexian.com/services/debian-lts.html
Learn to master Debian: http://debian-handbook.info/get/

**Comment by jbouse** [comment]
@rhertzog Yes, the salt package are "Arch: all" but there are "Arch: amd64" dependency packages in the repository which wouldn't work. I think I've figured a better way than my initial patch for map.jinja that doesn't have any adverse effect on Debian/Ubuntu but fixes the pkgrepo/key_url for Rasbian systems. 

The remaining piece I'm working on is the salt.pkgrepo include

**Comment by jbouse** [comment]
Got the PR submitted which currently corrects the pkgrepo and key_url values to match documented URLs for repo.saltstack.com for Raspbian... I still haven't found a more elegant way than to symlink the salt/pkgrepo/debian as salt/pkgrepo/raspbian... My attempts at tweaks to the salt/pkgrepo/init.sls have all failed at this point so far.

**Comment by jbouse** [comment]
I believe I got a solid solution and it also improves/fixes Ubuntu as well. I believe the salt/pkgrepo/ubuntu/ is actually no longer needed now as well but I left it alone.
